src/inspect_ai/log/_log.py

```diff
@@ -646,6 +647,53 @@ def rich_traceback(
     return rich_tb
 
 
+def truncate_traceback(
+    exc_type: Type[Any],
+    exc_value: BaseException,
+    exc_traceback: TracebackType | None,
+    max_length: int = 1048576,  # 1MB
+) -> Tuple[str, bool]:
+    tb_list = traceback.format_exception(exc_type, exc_value, exc_traceback)
+
+    # Keep the front and back of the traceback
+    header = tb_list[0]
+    error_msg = tb_list[-1]
+
+    # Join the middle parts (stack frames)
+    frames = "".join(tb_list[1:-1])
+
+    # It all fits, use it as is
+    full_tb = header + frames + error_msg
+    if len(full_tb) <= max_length:
+        return full_tb, False
+
+    ellipsis = "\n...\n"
+
+    # Minimum header size
+    header_size = min(len(header), 512)
+
+    # Minimum frames size
+    min_frames_size = min(len(frames), 2048)
+
+    # Remaining space for error message
+    error_msg_size = max(0, max_length - header_size - min_frames_size - len(ellipsis))
+
+    # Truncate each part as needed
+    truncated_header = header[:header_size]
+    truncated_frames = frames[:min_frames_size]
+    truncated_error = error_msg[:error_msg_size]
+
+    # Add ellipsis to any truncated part
+    if len(header) > header_size:
+        truncated_header += "..."
+    if len(frames) > min_frames_size:
+        truncated_frames += ellipsis
+    if len(error_msg) > error_msg_size:
+        truncated_error = truncated_error + "..."
```

Comment:
I would suggest including both the start and end of the traceback, as well as the start and end of the error message
```suggestion
    # Truncate each part as needed
    def truncate(s: str, sz: int, ellipsis: str):
        if sz <= len(s): return s
        sz = min(sz // 2, len(s) // 2)
        return f"{s[:sz]}{ellipsis}{s[-sz:]}"

    truncated_header = truncate(header, header_size, " [...] ")
    truncated_frames = truncate(frames, min_frames_size, ellipsis)
    truncated_error = truncate(error_msg, error_msg_size, " [...] ")
```

Reply:
Thx for suggestion, improved version!